### PR TITLE
[REVIEW] enable per-thread default stream in gpu ci

### PR DESCRIPTION
**Pull request by rongou**
Always build and test with per-thread default stream enabled, as discussed in #5614.

@harrism @jrhemstad @kkraus14

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -92,9 +92,9 @@ conda list
 
 logger "Build libcudf..."
 if [[ ${BUILD_MODE} == "pull-request" ]]; then
-    $WORKSPACE/build.sh clean libcudf cudf dask_cudf benchmarks tests
+    $WORKSPACE/build.sh clean libcudf cudf dask_cudf benchmarks tests --ptds
 else
-    $WORKSPACE/build.sh clean libcudf cudf dask_cudf benchmarks tests -l
+    $WORKSPACE/build.sh clean libcudf cudf dask_cudf benchmarks tests -l --ptds
 fi
```
Comment (kkraus14) [inline ci/gpu/build.sh:98]: @dillon-cullinan I assume the build mode is "pull-request" when CI is running against a PR. What is the `else` clause here used?

Comment (dillon-cullinan) [reply ci/gpu/build.sh:98]: For `branch` builds (merge commits + nightlies), we make sure to build the legacy code as well.

Comment (kkraus14) [reply ci/gpu/build.sh:98]: Got it. Legacy code is gone so we can likely remove that option, but seems like this is good to go then. Thanks!

**Comment by kkraus14** [comment]
Waiting to merge until tests have been run with RMM with the new PTDS enabled

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**